src/test/run-make/issue-11908/Makefile

```diff
@@ -12,12 +12,11 @@ all:
 	$(RUSTC) foo.rs --crate-type=dylib
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
-	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+	# dylibs fail with strange errors, but not assertions.
+	! $(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | grep "assertion failed"
```

Comment:
It *should* generate no error, but I'm in no way equipped to debug the error. It doesn't seem related to my change though -- even if you only had one copy this would fail.